src/cycles/simple_cycles.c

```diff
@@ -291,9 +295,11 @@ static igraph_error_t simple_cycles_circuit(
         IGRAPH_ALLOW_INTERRUPTION();
     }
 
-    IGRAPH_ASSERT(igraph_stack_int_size(&v_stack) == 0);
-    IGRAPH_ASSERT(igraph_stack_int_size(&e_stack) == 0);
-    IGRAPH_ASSERT(igraph_stack_int_size(&neigh_iteration_progress) == 0);
+    if (! state->stop_search) {
+        IGRAPH_ASSERT(igraph_stack_int_size(&v_stack) == 0);
+        IGRAPH_ASSERT(igraph_stack_int_size(&e_stack) == 0);
+        IGRAPH_ASSERT(igraph_stack_int_size(&neigh_iteration_progress) == 0);
+    }
```

Comment:
@GenieTim In particular, I believe these was a bug here, and these assertions are only valid if the search was not interrupted. Does this make sense to you?